### DISCUSSION TITLE
PHPMailer - Limit email attachments by file size and file type

### DISCUSSION THREAD

**Discussion by Zigii** [PHPMailer/PHPMailer#2880] [Q&A] [answered]
Hi,

Using PHPMailer with a form to send email attachment.

I want to do this securely and safely but also limit attachment file size and file type.

Looks like the example file at:
https://github.com/PHPMailer/PHPMailer/blob/master/examples/send_multiple_file_upload.phps
... is the way to go.

I've got the basic file working but now trying to adapt it to limit attachment size and file type.

See my updated file below (only relevant section of script is shown). 

What have I done wrong? Any help appreciated!

 `//Create a message
   
    $mail->setFrom('some@emailaddress.com', 'First Last');
    $mail->addAddress('another@emailaddress.com', 'John Doe');
    $mail->Subject = 'PHPMailer file sender';
    $mail->Body = 'My message body';
    //Attach multiple files one by one
    for ($ct = 0, $ctMax = count($_FILES['userfile']['tmp_name']); $ct < $ctMax; $ct++) {
        //Extract an extension from the provided filename
        $ext = PHPMailer::mb_pathinfo($_FILES['userfile']['name'][$ct], PATHINFO_EXTENSION);
        //Define a safe location to move the uploaded file to, preserving the extension
        $uploadfile = tempnam(sys_get_temp_dir(), hash('sha256', $_FILES['userfile']['name'][$ct])) . '.' . $ext;
        $filename = $_FILES['userfile']['name'][$ct];
		
		// restrict upload to 2 MB
		$maxsize = 2 * 1024 * 1024; 
		// allowed mime-types
		$types = array('application/pdf', 'application/msword', 'application/vnd.openxmlformats-officedocument.wordprocessingml.document', 'application/vnd.ms-excel', 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet');
		
		if(filesize($uploadfile) < $maxsize && in_array(mime_content_type($uploadfile),$types)){
       		move_uploaded_file($_FILES['userfile']['tmp_name'][$ct], $uploadfile); 
		}
		
		
       	$mail->addAttachment($uploadfile, $filename);
       	$mail->send();
	}`

**Comment by Synchro** [comment, marked as answer]
You also asked this question [on Stack Overflow](https://stackoverflow.com/questions/75658282/phpmailer-limit-email-attachments-by-file-size-and-file-type), and the comments there tell you what you're doing wrong. In particular, you're attempting to add the files regardless of whether they match your criteria or not, and you're sending the message every time around the loop, which will result in multiple messages, each one with one more attachment than the last. Look at how the original script you based this on works.

**Reply by Zigii** [reply]
Thanks for your comments Synchro. I'll rebuild the script.

Yes, I asked this question on Stack Overflow. I originally read through the PHPMailer documentation and tested a few example scripts on Github. However, I didn't realise that Github offered a Q&A forum so question was posted on Stack Overflow.
Later, back on Github, I finally noticed the Q&A and figured Github/PHPMailer might be a better place to ask the question. Apologies for any inconvenience.

Anyhow, once again, many thanks and all the best!
